fix(migrations): delete barrel exports in standalone migration

crisbeto opened the pull request:
Adds some logic to automatically delete `export * from './foo'` style imports. Previously they weren't being picked up, because finding all the references using the language service doesn't include barrel exports.

devversion commented on packages/core/schematics/ng-generate/standalone-migration/prune-modules.ts:
```diff
@@ -55,10 +66,26 @@ export function pruneNgModules(
+    for (const node of nodes) {
+      const sourceFile = node.getSourceFile();
+
+      if (!filesToRemove.has(sourceFile) && canRemoveFile(sourceFile, nodes)) {
```
Is `filesToRemove` ever populated at this point? seems like it would always be empty unless I miss something.

crisbeto replied:
Good catch, I actually meant to add to it here. Changed.

crisbeto committed:
fix(migrations): delete barrel exports in standalone migration

Adds some logic to automatically delete `export * from './foo'` style imports. Previously they weren't being picked up, because finding all the references using the language service doesn't include barrel exports.
